plugin-lifecycle/src/main/java/com/mapbox/maps/plugin/lifecycle/MapboxLifecyclePluginImpl.kt

```diff
@@ -0,0 +1,75 @@
+package com.mapbox.maps.plugin.lifecycle
+
+import android.content.ComponentCallbacks
+import android.content.res.Configuration
+import android.view.View
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleObserver
+import androidx.lifecycle.OnLifecycleEvent
+import androidx.lifecycle.ViewTreeLifecycleOwner
+import com.mapbox.common.Logger
+import com.mapbox.maps.MapboxLifecycleObserver
+import com.mapbox.maps.plugin.delegates.MapPluginProviderDelegate
+
+/**
+ * Concrete implementation of MapboxLifecyclePlugin.
+ */
+class MapboxLifecyclePluginImpl : MapboxLifecyclePlugin {
+  /**
+   * Register a MapboxLifecycleObserver to observe life cycle events from LifecycleOwner
+   *
+   * @param mapView the instance of mapView, will get the LifecycleOwner from mapview's parent
+   * @param observer the observer that listen to the life cycle events
+   */
+  override fun registerLifecycleObserver(mapView: View, observer: MapboxLifecycleObserver) {
+    val lifecycleOwner = ViewTreeLifecycleOwner.get(mapView)
+    if (lifecycleOwner == null) {
+      Logger.e(
+        TAG,
+        """Can't get lifecycleOwner for mapview,
+          please make sure the host Activity is AppCompatActivity and the version of appcompat is 1.3.0+.
```

Comment:
why appcompat 1.3.0?

Reply:
`ViewTreeLifecycleOwner.get(mapView)` will null and we can't hook into the lifecycle events.

Reply:
This seems to be important, we should state it clearly in the readme and maybe the separate lifecycle guide.

Reply:
Can we maybe add a lint error somewhere to validate it?

Reply:
Yeah, already added in the readme. I'm currently working on adding the lint check for lifecycle events, however not sure whether could check library version.

Reply:
It would be good to have it in the migration guide, as well as maybe a separate guide?